lib/msf/core/payload/windows/encrypted_reverse_tcp.rb

```diff
@@ -59,7 +59,7 @@ def generate(opts={})
       opt_lvl:       datastore['OptLevel'],
       keep_src:      datastore['KeepSrc'],
       keep_exe:      datastore['KeepExe'],
-      f_name:        (staged? ? 'reverse_pic_stager.exe' : 'reverse_pic_stageless.exe'),
+      f_name:        Tempfile.new(staged? ? 'reverse_pic_stager.exe' : 'reverse_pic_stageless.exe').path,
```

Comment:
The initial change resulted in a file name like this: `reverse_pic_stageless.exe20191211-99903-1qxrawr`. Using an array will force the extension if that's preferred.

```suggestion
      f_name:        Tempfile.new([ staged? ? 'reverse_pic_stager' : 'reverse_pic_stageless', '.exe' ]).path,
```

Reply:
Thanks, I opted to avoid pushing the extension here and instead let mingw.rb handle adding it itself.